logstash-core/lib/logstash/plugin.rb

```diff
@@ -122,12 +123,22 @@ def metric
                          LogStash::Instrument::NamespacedNullMetric.new(@metric, :null)
                        end
   end
+
   # return the configured name of this plugin
   # @return [String] The name of the plugin defined by `config_name`
   def config_name
     self.class.config_name
   end
 
+  # commit event to dlq
+  def dlq_commit(event, reason)
```

Comment:
> could not be deserialized using the avro-codec upon input from kafka

In this scenario, I would say we should reject this and leave it in kafka? 

My thought with the DLQ is that we put undeliverable messages into it with the idea that human intervention could make a correction and it could then be delivered. With an avro decoding failure (because someone is sending garbage bytes, maybe?), I don't know what kind of human intervention we could expect to resolve this?